### PR TITLE
Add PSK support

### PR DESCRIPTION
**Pull request by ndokmai**
Add PSK support to the library with one basic test.

### CODE REVIEW

**Comment by lachlansneff** [comment]
I'd really like to see this make its way in. Is there a path forward here?

**Comment by jethrogb** [comment]
Needs review @alexanderheuts

File: mbedtls/src/ssl/config.rs [hunk 1]
```diff
@@ -329,6 +329,49 @@ impl<'c> Config<'c> {
             )
         }
     }
+
+    /// psk and psk_identity cannot be empty
+    pub fn set_psk(&mut self, psk: &[u8], psk_identity: &str) -> Result<()> {
+        assert!(psk_identity.len()>0);
+        assert!(psk.len()>0);
```
Comment (raoulstrackx) [inline mbedtls/src/ssl/config.rs:336]: The `mbedtls_ssl_conf_psk` function already checks these arguments. I don't see a reason to check them here again

File: mbedtls/src/ssl/config.rs [hunk 2]
```diff
@@ -329,6 +329,49 @@ impl<'c> Config<'c> {
...
+    {
+        unsafe extern "C" fn psk_callback<F>(
+            closure: *mut c_void, 
+            ctx: *mut ssl_context, 
+            psk_identity: *const c_uchar, 
+            identity_len: size_t) -> c_int 
+            where 
+                F: FnMut(&mut HandshakeContext, &str) -> Result<()>,
+        {
+            assert!(identity_len>0);
```
Comment (raoulstrackx) [inline mbedtls/src/ssl/config.rs:356]: Same as above, I don't see a reason for the assert.

File: mbedtls/src/ssl/context.rs [hunk 3]
```diff
@@ -205,6 +205,16 @@ impl<'ctx> HandshakeContext<'ctx> {
                 .map(|_| ())
         }
     }
+
+    /// psk cannot be empty
+    pub fn set_psk(&mut self, psk: &[u8]) -> Result<()> {
+        assert!(psk.len()>0);
```
Comment (raoulstrackx) [inline mbedtls/src/ssl/context.rs:211]: Same as above, I don't see a reason for the assert.

File: mbedtls/tests/ssl_conf_psk.rs [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+use support::entropy::entropy_new;
+
+use mbedtls::rng::CtrDrbg;
+use mbedtls::ssl::config::{Endpoint, Preset, Transport};
+use mbedtls::ssl::{Config, Context, HandshakeContext};
+use mbedtls::Result as TlsResult;
+
+
+fn client(mut conn: TcpStream, psk: &[u8]) -> TlsResult<()> {
+    {
```
Comment (raoulstrackx) [inline mbedtls/tests/ssl_conf_psk.rs:16]: Nitpick: this doesn't need to be in it's own block

**Comment by Taowyoo** [comment]
Close this PR since this is outdated and PSK support is already merged.
Pls check comments in https://github.com/fortanix/rust-mbedtls/pull/205 for more info

**Pull request closed without merging**